service/history/workflow/mutable_state.go

```diff
@@ -362,6 +364,8 @@ type (
 		)
 		UpdateActivity(*persistencespb.ActivityInfo) error
 		UpdateActivityWithTimerHeartbeat(*persistencespb.ActivityInfo, time.Time) error
+		UpdateActivityWithCallback(*persistencespb.ActivityInfo, UpdateActivityCallback) error
```

Comment:
Does this need to be exported?

Reply:
yes, it is used outside on mutable_state_impl.go